chain4travel/camino-signavault
# feature/fix for pending tx

- TECH-506 Initial implementation for signature collection service
- TECH-506 Created migrations scripts for multisig schema, docker compose improvements
- TECH-506 Updates to create and query multisig
- TECH-506 Removed dbr package and refactored queries using go sql package instead
- TECH-506 Changes to support getting one transaction for multisig alias
- TECH-506 Added method and api endpoint to insert new signer
- TECH-506 externalized configuration to yaml file
- TECH-506 Changes for integration testing
- TECH-506 Updates on integration tests
- TECH-506 Refactoring according to new specs
- TECH-506 Implemented update multisig tx, added validations, various improvements
- TECH-506 Added Dockerfile, minor improvements and cleanup
- TECH-506 Added placeholders for mysql connection details
- add build docker workflow
- Refactored endpoints to use database id as unique identifier
- Added http post call to get msig alias info from api node
- Spec changes - Use separate struct for creating msig alias - Unified Signers with Owner structs - Owners & threshold are fetched from api node and saved in db - Creator of msig alias is also the first single signer - Removed unnecessary code and disabled validations for now.
- Implementation of signer authorization
- Added camino node url in config file
- Updates on complete tx operation
- Added extraction of owner address from signature & other improvements
- Updated git workflow to initialize git submodules
- Added auth on get all by alias endpoint
- Get multisig tx test
- Added auth check on sign and complete tx, various improvements
- Fix for duplicate owners on multisig tx creation
- Added multisig tx dao for database operations
- Fix signature verification on signing and completing tx - Signing takes the tx database id and timestamp as signature params - Completing takes the tx database id, timestamp and actual transaction id (generated by node) as signature params
- Reduced dependency decoupling between services for easier testing
- Tx verification on chain
- Fixed add signer operation
- Moved error messages in multisig service to global variables
- Exposed multisig service with an interface
- Change signature format of signTx endpoint
- Test refactoring
- Gomock integration and test improvements
- Added output owners on tx, error handling improvements
- UnsignedTx is hashed and used as primary key
- Added check for not allowing multiple pending transactions for the same alias
- Increased unsigned_tx size to 1024
- Replace completeTx endpoint with issueTx ones and update deps
- Marshaling and unmarshaling fixes of tx and utx
- Linter suggestion fixes
- Fixed create multisig test
- Update txID only if issueTx is successful
- Added license file and updated copyright
- Typescript api client implementation (#3)
- Add metadata field to multisigTx (#4)
- Added multisig dao integration tests using testcontainers
- Add signavaultjs as npm dependency and add register node tx example (#6)
- use local sql and add signavault to docker-compose file (#7)
- Added docker compose file for local development
- Add validator transaction example and signavaultjs typings (#8)
- Added cors policy to allow all origins
- Add Readme file (#11)
- Added timestamp in multisig_tx & minor fixes
- Update README.md
- Generate new client based on timestamp addition (#14)
- Added build & reset db scripts
- add config_template.yml (#16)
- Fix for failing tests after timestamp field addition
- [Service] Generate ID by TX instead UnsignedTx
- Getting all multisig tx for alias returns an empty array instead of an error in case alias does not exist
- Fix for test after tx id generation change
- Fix for db timezone change affecting timestamp
- Fix for bad error handling when get a transaction with wrong id
- Handler tests and dependency injection improvements
- Added expiration date for transactions
- Generate new client based on expiration data addition and adjust example (#25)
- Added default configurable tx expiration date
- Updated config template with default expiration days
- Added chainId property on multisig tx
- Fix for pending check taking into consideration the expiration time
